assets/src/apps/authoring/components/AdvancedAuthoringModal.tsx

```diff
@@ -0,0 +1,26 @@
+import { Modal, ModalProps } from 'react-bootstrap';
+import React, { useContext, useState } from 'react';
+
+//  This helps us target our react-bootstrap modals to be within the shadow-dom that the advanced authoring app lives.
+const ModalContext = React.createContext<HTMLDivElement | null>(null);
+
+export const ModalContainer: React.FC = ({ children }) => {
+  const [modalContainer, setModalContainer] = useState<HTMLDivElement | null>(null);
+
+  return (
+    <div>
+      <ModalContext.Provider value={modalContainer}>{children}</ModalContext.Provider>
+      <div id="advanced-authoring-modals" ref={setModalContainer}></div>
+    </div>
+  );
+};
+
+export const AdvancedAuthoringModal: React.FC<ModalProps> = (props) => {
```

Comment:
Use this instead of the plain <Modal> and it will all magically work inside the shadow-dom